src/core/u-compress.c

```diff
@@ -355,9 +406,10 @@ REBSER *Decompress(
 
     // Since the initialization succeeded, go ahead and make the output buffer
     //
-    REBSER *output = Make_Binary(buf_size);
+    REBSER *output = Make_Binary(sizeof(REBSER*) + buf_size);
+    *cast(REBSER**, BIN_HEAD(output)) = output; // self-referential prefix
     strm.avail_out = buf_size;
-    strm.next_out = BIN_HEAD(output);
+    strm.next_out = BIN_HEAD(output) + sizeof(REBSER*);
 
```

Comment:
Maybe introducing another series flag and set it when Rebol_Malloc/Rebol_Malloc_Setup/Prepare, and then Rebol_Realloc knows if the bias includes a sizeof(REBSER*)? Perhaps it's a good idea to have such a flag anyway, so that various checking can be done, e.g. it couldn't be copied in a usual way.